perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp

```diff
@@ -174,14 +175,16 @@ void RoiClusterFusionNode::fuseOnSingleImage(
     double max_iou = 0.0;
     bool is_roi_label_known =
       feature_obj.object.classification.front().label != ObjectClassification::UNKNOWN;
+    bool is_long_range_obj = get_object_square_distance(feature_obj) >
+                             iou_x_use_distance_threshold_ * iou_x_use_distance_threshold_;
     for (const auto & cluster_map : m_cluster_roi) {
       double iou(0.0), iou_x(0.0), iou_y(0.0);
       if (use_iou_) {
         iou = calcIoU(cluster_map.second, feature_obj.feature.roi);
       }
       // use for unknown roi to improve small objects like traffic cone detect
       // TODO(badai-nguyen): add option to disable roi_cluster mode
-      if (use_iou_x_ || !is_roi_label_known) {
+      if (use_iou_x_ || !is_roi_label_known || is_long_range_obj) {
```

Comment:
In case of the long range object has IoU score > 0, the total score (= `iou` + `iou_x`) might be bigger than its threshold even if it is noise.

Reply:
@ktro2828 You're right, I  fixed in https://github.com/autowarefoundation/autoware.universe/pull/5148/commits/3ae96c0a0e82f01daec611776a70e089f497cc80

Reply:
I think this is not enough. In case with both `use_iou = use_iou_x = true`, the `iou_threshold` would be set to greater than 1.0. And then, the long range objects are always  filtered out.

```cpp
// Use IoUX for the long range objects and detected unkown objects such as traffic cone instead of IoU
if (use_iou_)
{
  iou = (is_long_range_obj || !is_roi_label_known) ? calcIoUX(cluster_map.second, feature_obj.feature.roi)
                                                   : calcIoU(cluster_map.second, feature_obj.feature_roi)
}

if (use_iou_x_)
{
  iou_x = calcIoUX(cluster_map.second, feature_obj.feature.roi);
}
```

Do you want to always handle total IoU score with `iou + iou_x` for detected unknown objects such as traffic cone?

Reply:
@ktro2828 Thank you for your comment. You're right. 
> In case with both use_iou = use_iou_x = true,

Anyways, because of  this https://github.com/autowarefoundation/autoware.universe/blob/3ae96c0a0e82f01daec611776a70e089f497cc80/perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp#L198 , if `use_iou = use_iou_x = true,` the max_iou might be greater than 1.0 not only for long_range_obj case only. 
So, IMO in the document should mention to use only one of `use_iou`, `use_iou_x` or `use_iou_y` or they are should block each other somewhere.

> Do you want to always handle total IoU score with iou + iou_x for detected unknown objects such as traffic cone

No. It's my mistake. I will fixed.

Reply:
@ktro2828 I changed to single iou mode selection and fix `iou_x` use for unknown object in https://github.com/autowarefoundation/autoware.universe/pull/5148/commits/7a064470d05e59876d73bfbde598b1ff0d175411

Reply:
@yukke42 Do you have any concerns that it only supports one IoU calculation method?

Reply:
@ktro2828 @badai-nguyen 
No.